ReactiveCocoa/Swift/Property.swift

```diff
@@ -17,17 +20,26 @@ public protocol PropertyProtocol {
 	/// completes when the property has deinitialized, or has no further
 	/// change.
 	var signal: Signal<Value, NoError> { get }
+
+	/// The property sources of `self`, or `nil` if `self` owns the
+	/// lifetime of its signal and producer.
+	var sources: [AnyObject]? { get }
 }
 
 /// Represents an observable property that can be mutated directly.
-///
-/// Only classes can conform to this protocol, because instances must support
-/// weak references (and value types currently do not).
-public protocol MutablePropertyProtocol: class, PropertyProtocol {
+public protocol MutablePropertyProtocol: PropertyProtocol {
 	/// The current value of the property.
 	var value: Value { get set }
 }
 
+extension PropertyProtocol {
+	/// The property sources of `self`, or `nil` if `self` owns the
+	/// lifetime of its signal and producer.
+	public var sources: [AnyObject]? {
```

Comment:
The `PropertyProtocol.sources` requirement in `PropertyProtocol` is to replace the `AnyProperty.capture` using runtime dynamic cast. Since the protocol is public, this have to be public too. Moreover, custom property types should be allowed to specify their sources.

Though whether or not a default implementation should be supplied is debatable.